### DISCUSSION TITLE
IAGL Discussions

### DISCUSSION THREAD

**Discussion by zach-morris** [zach-morris/plugin.program.iagl#181] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Phyxsius7** [comment]
Hi, mist this part. A bit about me.
I go understand the alias Phyxsius7, am a very long time user of Kodi. To give you an idea, my first installation of it was back them when it was called XBMC. 
Also a proud owner of one of the first batch of Sega Saturn console. Have a respectable collection Sega Saturn games and accessories.
Still functional and would like to keep it that way. 
So, IAGL is a very welcome solution.

**Comment by Hi-Phile** [comment]
Hi, i'm not a new user to Kodi, but not very experienced in how it can be deployed/setup/modified.  I'm using Kodi 19.4 from a flathub on Arch Linux (RebornOS).  Kodi does work, and have been able to use it for videos, and music.  When I went to install the IAGL, I was able to do that and IAGL loads up no problems, I even linked my account login via the settings for IAGL.  

My problem is that I cannot get any game to launch, although it does download.  The dialog box that appears is "The game isn't compatible with any available emulators."

So, I then attempt to configure the external launcher 'retroarch' and using the appropriate libretro-<emulator>.  But, what I see is that because this is kodi from a flatpak, it's not in the standard install locations.  That shouldn't be an issue, just configure the path to retroarch and retroarch.cfg,  One problem I encountered was that because it's a flatpak it's in a sandbox environment, so what I did was create sym-links to retroarch and retroarch.cfg because without the sym-link the IAGL or Kodi can't 'see' the files outside of the sandbox.

I placed the sym-links under /home/user/.var/app/tv.kodi.Kodi/config.  Then once I did that IAGL was able to save the settings.

But, the problem with it not launching with the same error.  "The game isn't compatible with any available emulators."

I'd appreciate any tips or suggestions in order to get the IAGL to work in flatpak Kodi environment.

If you want to check out the kodi.log, it's [https://termbin.com/y8t9](url)

**Comment by Hi-Phile** [comment]
I am adding screenshots below, but no matter how I attempt to get any game ROM to launch,  I keep getting the error in IAGL:

Failed to play game - This game isn't compatible with any emulators

I attempted to configure the path for external launchers, but due to this being in a sandbox environment you cannot select the retroarch program, but what was able to do was create symlinks to the files.

Not sure if this is due to flatpak, but in any event it does not seem to launch retroarch either within Kodi and from IAGL Program Addon which I was able to successfully install.  The games actually download, but for whatever the reason, there's no emulator.

Kodi.log is available at termbin.com/y8t9

Also, when I launch the retroarch outside of Kodi, I am able to play a ROM with the Xbox One S controller and the controller also works in Kodi.

Hopefully, it's something simple that I'm missing.
![Screenshot from 2022-08-19 16-39-19](https://user-images.githubusercontent.com/35350630/185720029-fcd35dc0-b9f7-466e-b474-cd42f6f8904e.png)
![Screenshot from 2022-08-19 16-37-19](https://user-images.githubusercontent.com/35350630/185720142-1d72ee47-3825-4517-941c-fbaed68df8a7.png)
![Screenshot from 2022-08-19 16-38-31](https://user-images.githubusercontent.com/35350630/185720275-df0a055e-1127-4f39-be9a-33669a614876.png)
